store/account.go

```diff
@@ -1006,7 +1006,7 @@ func (a *Account) MessageReader(m Message) *MsgReader {
 func (a *Account) Deliver(log *mlog.Log, dest config.Destination, m *Message, msgFile *os.File, consumeFile bool) error {
 	var mailbox string
 	rs := MessageRuleset(log, dest, m, m.MsgPrefix, msgFile)
-	if rs != nil {
+	if rs != nil && rs.Mailbox != "" {
```

Comment:
i don't think Mailbox can be empty. the is a required field according to config/config.go, meaning it cannot be empty.

Reply:
It was, this PR makes it optional so that the filter doesn't have to reroute the mail.